### PR TITLE
promote e2e tests for taint-tolerations in predicates

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Promote e2e test for taint-tolerations in predicates to conformance. This was lgtm'd in another PR but was abandoned: https://github.com/kubernetes/kubernetes/pull/78240

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Conformance promotion of this test:

"validates that taints-tolerations is respected if matching"
--- Takes approximately 36 seconds for successful execution.
--- Verifies that the pod is scheduled when the pod's toleration matches the node taint.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/area conformance
/priority important-longterm
/sig scheduling
/sig apps
/cc johnbelamaric

### CODE REVIEW

File: test/e2e/scheduling/predicates.go [hunk 1]
```diff
@@ -491,11 +491,14 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
...
-	// 3. Try to relaunch the pod with tolerations tolerate the taints on node,
-	// and the pod's nodeName specified to the name of node found in step 1
-	ginkgo.It("validates that taints-tolerations is respected if matching", func() {
+	/*
+		Release : v1.17
+		Testname: Scheduler, taints-tolerations matching
+		Description: After removing the pod form the available node, verify that the
+		pod (with toleration against node taint) is scheduled when the node is tainted with a
+		random taint. The pod MUST be scheduled onto the node
+	*/
```
Comment (Huang-Wei) [inline test/e2e/scheduling/predicates.go]: The wording here is a bit unclear to me.

The basic idea here is: Find an available node which can afford the testing pod, and taint that node with a specific key with effect NoSchedule, then schedule a Pod with (1) particular nodeLabel to ensure it can only be placed onto node, and also (2) particular tolerations spec to ensure it matches the pre-applied taint. Finally check if the Pod is running on the node.

And also, L529~L533 may be a copy/paste mistake, we need to reword them as well.

Could you reword these 2 places?

**Comment by johnbelamaric** [comment]
This taints a node, which I believe may in turn cause evictions. As such it should be marked as `[Disruptive]`

File: test/e2e/scheduling/predicates.go (hunk 1 above)
Comment (Jefftree) [reply test/e2e/scheduling/predicates.go]: ack, reworded the description. Sorry what is the issue with L529-533? It seems to be doing the same checks as the tests above it?

Comment (Huang-Wei) [reply test/e2e/scheduling/predicates.go]: Nvm, L529~L533 is fine. Just leave them as is.

Comment (Jefftree) [reply test/e2e/scheduling/predicates.go]: Just did a rebase, would you mind reviewing?

**Comment by Huang-Wei** [comment]
/lgtm
on scheduling side.

**Review by spiffxp — CHANGES_REQUESTED** [review]
ref: https://github.com/kubernetes/kubernetes/pull/78240/files#r309937152

The comment at the bottom of this test about a race condition is incorrect per https://github.com/kubernetes/kubernetes/pull/78240/files#r308380842 and I'd rather get rid of the copy paste.

The pod is verified to go to Running (not just NotPending) at the start of the test, I think it's reasonable to do the same check at the end of this this test.

ie: instead of
`framework.ExpectNoError(e2epod.WaitForPodNotPending(cs, ns, tolerationPodName))`
consider using
`framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(cs, ns, tolerationPodName))`

**Comment by johnbelamaric** [comment]
Sigh. 

/approve

**Comment by BenTheElder** [comment]
AFAICT https://github.com/kubernetes/kubernetes/issues/86379 is failing in GCE & KIND 100% of the time since we promoted it.

File: test/e2e/scheduling/predicates.go [hunk 2]
```diff
@@ -485,12 +488,7 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 			NodeSelector: map[string]string{labelKey: labelValue},
 		})
 
-		// check that pod got scheduled. We intentionally DO NOT check that the
-		// pod is running because this will create a race condition with the
-		// kubelet and the scheduler: the scheduler might have scheduled a pod
-		// already when the kubelet does not know about its new taint yet. The
-		// kubelet will then refuse to launch the pod.
-		framework.ExpectNoError(e2epod.WaitForPodNotPending(cs, ns, tolerationPodName))
+		framework.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(cs, ns, tolerationPodName))
```
Comment (spiffxp) [inline test/e2e/scheduling/predicates.go]: This change should have been split out into a separate PR prior to promoting to Conformance, so we could verify whether the test still passed

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/kubernetes/pull/86386 to revert

**Comment by johnbelamaric** [comment]
Yep. Sorry about that.

File: test/e2e/scheduling/predicates.go (hunk 2 above)
Comment (Jefftree) [reply test/e2e/scheduling/predicates.go]: ack, sorry about that
